src/shared/hooks/useCanAccountUnstakeAmount.tsx

```diff
@@ -1,28 +1,74 @@
 import { BackersManagerAbi } from '@/lib/abis/v2/BackersManagerAbi'
 import { BACKERS_MANAGER_ADDRESS } from '@/lib/constants'
 import { useAccount, useReadContract } from 'wagmi'
+import { parseEther } from 'viem'
 
-export const useCanAccountUnstakeAmount = (amount: bigint) => {
+/**
+ * Custom hook to determine if an account can unstake a specified amount of RIF tokens
+ *
+ * @param amount - The amount to unstake in bigint format
+ * @param stRifBalance - The current stRIF balance as a string
+ * @returns An object containing:
+ *  - canAccountWithdraw: boolean indicating if the account can withdraw the specified amount
+ *  - isCanAccountWithdrawLoading: boolean indicating if the data is still loading
+ *
+ * @remarks
+ * The hook calculates if an account can unstake by:
+ * 1. Fetching the backer's total allocation from the BackersManager contract
+ * 2. Converting input amount and balance to proper format using parseEther
+ * 3. Calculating available balance by subtracting total allocation from current balance
+ * 4. Comparing requested unstake amount against available balance
+ *
+ * @example
+ * ```tsx
+ * const { canAccountWithdraw, isCanAccountWithdrawLoading } = useCanAccountUnstakeAmount(
+ *   "100", // amount to unstake
+ *   "1000" // current stRIF balance
+ * );
+ * ```
+ */
+export const useCanAccountUnstakeAmount = (amount: string, stRifBalance: string) => {
   const { address } = useAccount()
-  const {
-    data: canAccountWithdraw,
-    isLoading: isCanAccountWithdrawLoading,
-    refetch: refetchCanAccountWithdraw,
-  } = useReadContract(
+  const { data: backerTotalAllocation, isLoading: isBackerTotalAllocationLoading } = useReadContract(
     address && {
       abi: BackersManagerAbi,
       address: BACKERS_MANAGER_ADDRESS,
-      functionName: 'canWithdraw',
-      args: [address, amount],
+      functionName: 'backerTotalAllocation',
+      args: [address],
       query: {
-        enabled: false,
+        refetchInterval: 10000,
       },
     },
   )
 
+  const parsedAmount = parseEther(amount) ?? 0n
+  const parsedBalance = parseEther(stRifBalance) ?? 0n
+  const balanceThatCanBeWithdraw = parsedBalance - (backerTotalAllocation || 0n)
```

Comment:
@Freshenext I think it would be better to use a `useMemo` to just calculate `canAccountWithdraw`. In the past, we experienced a strange behaviour of `useReadContract` that was affected by not having a `useMemo` to perform calculation of the `useReadContracts` results